flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -168,6 +171,9 @@
 
 	private final CompletableFuture<Void> closeFuture;
 
+	@Nullable
+	protected BufferReceivedListener listener;
```

Comment:
longer variable name like `bufferReceivedListener` in case we add more listener later?

Reply:
In general it is better to give more specific semantic naming. Here i take it for simple also considering the usage in RemoteInputChannel via `inputGate.listener.notifyBarrierReceived` which seems not very long. I can also take the full naming.